### PR TITLE
LPDS 16 Popup component

### PR DESCRIPTION
**Pull request by naufal22hadi**
# Description, Motivation and Context

`LPPopupDialogFragment` is a custom dialog fragment used by calling `showCustomPopup` function.

# JIRA Ticket

[LDPS-16](https://lionparcel.atlassian.net/browse/LPDS-16?atlOrigin=eyJpIjoiM2U1OGNmMTZlMGEwNDQ3NzhjZmZmODA5ZjdjMTdjY2MiLCJwIjoiaiJ9)

# What is the destination branch of this PR?

- Master

# Screenshots

| **Description** | **Image** |
| --- | --- |
| Popup Text Only |  |
| Popup Text with Illustration |  |
| Popup Text with Button |  |
| Popup Text with Button and Illustration |  |

### CODE REVIEW

**Review by fadhilahsaputro — APPROVED** [review]

File: core/src/main/java/com/lionparcel/commonandroid/popup/LPPopupDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            }
+
+        fun lpPopupTextButtonIllustration(
+            title: String,
+            content: String,
+            image: Int,
+            primaryButtonText: String,
+            secondaryButtonText: String,
+            primaryButtonListener : View.OnClickListener? = null,
+            secondaryButtonListener : View.OnClickListener? = null
```
Comment (falih-mulyana) [inline core/src/main/java/com/lionparcel/commonandroid/popup/LPPopupDialogFragment.kt:69]: kalau dari Android dev yang lain gimana, guys? kalian lebih prefer pake nama function yang beda atau pake 1 nama function di-overload berkali-kali dengan beberapa parameter? @fadhilahsaputro @risanjayy @fajardarwis @riomanik @daniimulyana

Comment (fadhilahsaputro) [reply core/src/main/java/com/lionparcel/commonandroid/popup/LPPopupDialogFragment.kt:69]: pake 1 nama function di-overload berkali-kali ini mas

File: core/src/main/java/com/lionparcel/commonandroid/popup/LPPopupDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+        fun lpPopupTextOnly(title: String, content: String) =
+            LPPopupDialogFragment(PopupTypeModal.TEXT_ONLY).apply {
+                arguments = bundleOf(
+                    BASIC_TITLE_GENERAL to title,
+                    BASIC_SUB_TITLE_GENERAL to content
+                )
+            }
+
+        fun lpPopupTextIllustration(title: String, content: String, image: Int) =
```
[the remark this answers is not in the document]
Comment (naufal22hadi) [reply core/src/main/java/com/lionparcel/commonandroid/popup/LPPopupDialogFragment.kt:34]: Bisa sih mas, kalo pada mau gitu. Kayaknya lebih mudah juga pas dipakenya

[the remark this answers is not in the document]
Comment (falih-mulyana) [reply core/src/main/java/com/lionparcel/commonandroid/popup/LPPopupDialogFragment.kt:34]: oke mas @naufal22hadi , kalo gitu coba di-refactor lagi factory method-nya ya. Tengkyu.

[the remark this answers is not in the document]
Comment (naufal22hadi) [reply core/src/main/java/com/lionparcel/commonandroid/popup/LPPopupDialogFragment.kt]: done mas @falih-mulyana

File: core/src/main/java/com/lionparcel/commonandroid/popup/ShowCustomPopup.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+fun showCustomPopup(
+    fragmentManager: FragmentManager,
+    key: String,
+    dialogFragment: DialogFragment
+) {
+    fragmentManager.let {
+        dialogFragment.show(it, key)
+    }
+}
```
Comment (falih-mulyana) [inline core/src/main/java/com/lionparcel/commonandroid/popup/ShowCustomPopup.kt:14]: biasain di akhir file kasih 1 new line (enter) @naufal22hadi

File: core/src/main/java/com/lionparcel/commonandroid/popup/base/BaseDialogFragment.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    fun setWidthPercent(percentage: Int) {
+        val percent = percentage.toFloat() / 100
+        val dm = Resources.getSystem().displayMetrics
+        val rect = dm.run { Rect(0, 0, widthPixels, heightPixels) }
+        val percentWidth = rect.width() * percent
+        dialog?.window?.setLayout(percentWidth.toInt(), ViewGroup.LayoutParams.WRAP_CONTENT)
+    }
+
+}
```
Comment (falih-mulyana) [inline core/src/main/java/com/lionparcel/commonandroid/popup/base/BaseDialogFragment.kt:52]: no newline

File: sample/src/main/res/layout/activity_popup_component_sample.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    <Button
+        android:id="@+id/btntextButtonIllustration"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        style="@style/RedPaleSalmonButtonStyle"
+        android:text="POPUP TEXT BUTTON ILLUSTRATION"/>
+
+</LinearLayout>
```
Comment (falih-mulyana) [inline sample/src/main/res/layout/activity_popup_component_sample.xml:44]: eof

File: sample/src/main/java/com/lionparcel/commonandroidsample/popup/PopupComponentSampleActivity.kt [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                    title = "Title Insert Here",
+                    content = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. ",
+                    image = R.drawable.ill_badge_failed,
+                    primaryButtonText = "Primary Button",
+                    secondaryButtonText = "Secondary Button"
+                )
+            )
+        }
+    }
+}
```
Comment (falih-mulyana) [inline sample/src/main/java/com/lionparcel/commonandroidsample/popup/PopupComponentSampleActivity.kt:64]: eof

File: core/src/main/java/com/lionparcel/commonandroid/popup/res/values/styles-popup.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+</resources>
```
Comment (falih-mulyana) [inline core/src/main/java/com/lionparcel/commonandroid/popup/res/values/styles-popup.xml:3]: eof

File: core/src/main/java/com/lionparcel/commonandroid/popup/res/layout/lp_custom_popup_dialog_fragment.xml [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                android:textAppearance="@style/ButtonButton"
+                android:visibility="gone"
+                tools:visibility="visible"
+                tools:text="Secondary Button"/>
+
+        </LinearLayout>
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+</com.google.android.material.card.MaterialCardView>
```
Comment (falih-mulyana) [inline core/src/main/java/com/lionparcel/commonandroid/popup/res/layout/lp_custom_popup_dialog_fragment.xml:107]: eof

File: core/src/main/java/com/lionparcel/commonandroid/popup/res/drawable/bg_popup.xml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <solid android:color="@color/white"/>
+    <corners android:radius="@dimen/dp_12"/>
+</shape>
```
Comment (falih-mulyana) [inline core/src/main/java/com/lionparcel/commonandroid/popup/res/drawable/bg_popup.xml:6]: eof

**Review by falih-mulyana — CHANGES_REQUESTED** [review]
put newline at the end of each file

**Review by falih-mulyana — APPROVED** [review]
LGTM

**Pull request merged**